rasa/engine/storage/local_model_storage.py

```diff
@@ -88,6 +89,18 @@ def _extract_archive_to_directory(
     ) -> None:
         with TarSafe.open(model_archive_path, mode="r:gz") as tar:
             tar.extractall(temporary_directory)
+        LocalModelStorage._assert_not_rasa2_archive(temporary_directory)
+
+    @staticmethod
+    def _assert_not_rasa2_archive(temporary_directory: Union[Text, Path],) -> None:
```

Comment:
I checked and it's actually always a `Path` 
```suggestion
    def _assert_not_rasa2_archive(temporary_directory: Path) -> None:
```